godot-core/src/private.rs

```diff
@@ -41,24 +40,57 @@ sys::plugin_registry!(pub __GODOT_PLUGIN_REGISTRY: ClassPlugin);
 
 // Note: if this leads to many allocated IDs that are not removed, we could limit to 1 per thread-ID.
 // Would need to check if re-entrant calls with multiple errors per thread are possible.
-#[derive(Default)]
 struct CallErrors {
-    map: HashMap<i32, CallError>,
-    next_id: i32,
+    ring_buffer: Vec<Option<CallError>>,
+    next_id: u8,
+    generation: u16,
+}
+
+impl Default for CallErrors {
+    fn default() -> Self {
+        Self {
+            // [None; N] requires Clone.
+            ring_buffer: std::iter::repeat_with(|| None)
+                .take(Self::MAX_ENTRIES as usize)
+                .collect(),
```

Comment:
Just been going back through PRs i didnt get a chance to look at but just fyi you can do this now:
```rs
[const { None }; N]
```

Reply:
Oh, this is great to know, thanks! I was looking for alternative patterns but couldn't find this, maybe too recent :slightly_smiling_face: 

Will introduce as a drive-by-commit somewhere :+1:

Reply:
Unfortunately with `vec![...]` it doesn't work:

```rs
error: no rules expected the token `const`
  --> godot-core/src/private.rs:53:31
   |
53 |             ring_buffer: vec![const { None }; Self::MAX_ENTRIES as usize],
   |                               ^^^^^ no rules expected this token in macro call
   |
   = note: while trying to match end of macro
```

And directly `None` caused the original error:
```rs
error[E0277]: the trait bound `Option<call_error::CallError>: Clone` is not satisfied
    --> godot-core/src/private.rs:53:32
     |
53   |             ring_buffer: vec![ None; Self::MAX_ENTRIES as usize],
     |                          ------^^^^-----------------------------
     |                          |     |
     |                          |     the trait `Clone` is not implemented for `Option<call_error::CallError>`
     |                          required by a bound introduced by this call
     |
     = note: required for `Option<call_error::CallError>` to implement `Clone`
```

---

What does work is this, but then again that's not really better than the current code:
```rs
            ring_buffer: [const { None }; Self::MAX_ENTRIES as usize]
                .into_iter()
                .collect(),
```

Reply:
can't you just do `.into()`?

Reply:
I thought I tried that, but probably made a mistake. That works, thanks! :slightly_smiling_face: